### PR TITLE
Initial 2DUnet project

### PR DESCRIPTION
**Pull request by Edmondson-codes**
Created Project Base

### CODE REVIEW

**Comment by wangzhaomxy** [comment]
**<This is an initial inspection, no action is required at this point.>**

**File Organizing:** Wrong file path. All student-constructed documents should be packed in the ‘PatternAnalysis-2024/recognition/<student_constructed_folder>/’ folder. Please reorganize your files, otherwise it won't be merged into Shakes' branch.

**Problem Solving:**

- The algorithm can solve the problem. However, the model was just trained for two epochs, which is insufficient to train a model.
- Dice score in testing dataset: 0.9749. However, this number is the loss number instead of the Dice Coefficient, so the Dice score should be 1 - loss = 0.0351, which is extremely low due to training only two epochs leading to less fitting. 

**Model and functions:**

- It correctly uses Tensorflow to construct the 2D-UNet models and functions.
- No data augmentation.
- No validation observation: student just used original train/test dataset to train and test model instead of train/val/test.

**Code design:** Confused.
- Both train.py and tests.py contain the training and testing code. In addition, different definitions of dice coefficient appear in these two different scripts.
- The loading data function in dataset.py tends to load either images or masks controlled by the categorical argument, but both loading images and masks used the default argument. 
-  The code is hard to read.

**Code comment and docstring:**

- Minimal code comments 
- Minimal function docstrings

**Difficulty:** Easy.

**Additional Comments:**

- Minimal commits.
- Training only two epochs is inefficient for analysing the experiment.
-  Training/ validation/ testing plots are expected in the report.
-  No conclusion in the report.
-  No reference in the report.

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Adequate design and implementation. Code has some inconsistancies.</td><td>-1</td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>Header blocks. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>OK solution to problem. The algorithm can solve the problem. However, the model was just trained for two epochs.</td><td>-2</td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure NOT present. No predict.py</td><td>-1</td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting missing. </td><td>-1</td></tr>
	<tr><td></td><td>No Data leakage found. </td><td></td></tr>
	<tr><td></td><td>Difficulty : Easy. UNet2D (Easy Difficulty)</td><td>-10</td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Good Meaningful commit messages. </td><td></td></tr>
	<tr><td></td><td>Good Progressive commits. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Acceptable. Loks incomplete.</td><td>-2</td></tr>
	<tr><td></td><td>Model/technical explanation :Good. </td><td></td></tr>
	<tr><td></td><td>Description and Comments :Good. </td><td></td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. PDF checked.</td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Pull Request has problems: Submitted on time. Wrong branch.</td><td>-2</td></tr>
	<tr><td></td><td>Feedback action required: Feedback marks possible +2 if the requested changes are made. Change to correct branch and remove cache files for merge. </td><td>-2</td></tr>
	<tr><td></td><td>Request Description is good. </td><td></td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-21</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_

**Comment by gayanku** [comment]
Feedback marks possible +2 if the requested changes are made (see above).

**Comment by shakes76** [comment]
Does not solve the problem, training incomplete -3 additionally
